test-e2e/src/halo2/mod.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::HashMap;
```

Comment:
I don't think `test-e2e` should have more than 1 module (`lib.rs`). All other logic should be delegated to either `mopro-ffi`, or a custom prover (in this case `halo2_fibonacci`).